### ISSUE TITLE
Remove DXSDK JUN2010 dependency

### ISSUE THREAD

**Issue by Dwood15** [Dwood15/Tempera#19]
The dependencies holding us back are: `d3dx9.h` and `d3dx9math.h`

**Comment by BinToss** [comment]
Hey! I can't remember if Cpp projects support it, but I think you should know about the NuGet package for the DirectX 9 Extensions.
https://www.nuget.org/packages/Microsoft.DXSDK.D3DX

The compiler *should* be able to trim unused code objects during compile time, so I believe only the referenced code will be imported and the rest will be trimmed from the final assembly. Again, I'm more accustomed to the DotNet SDK compiler now, so I'm unsure if all (if any) C-C++ compilers support NuGet package imports and assembly trimming.

----

If you plan to deprecate dependencies on DirectX9 Extensions, then here are some good places to start:

> - https://walbourn.github.io/living-without-d3dx/
>   - https://walbourn.github.io/where-is-the-directx-sdk-2021-edition/
> - (see Note) https://docs.microsoft.com/en-us/windows/win32/direct3d9/d3dx
>   - https://walbourn.github.io/legacy-d3dx-on-nuget/
>   - https://www.nuget.org/packages/Microsoft.DXSDK.D3DX
> - https://github.com/microsoft/DirectXTK
>   - https://github.com/microsoft/DirectXMesh
>   - https://github.com/microsoft/DirectXShaderCompiler
>   - https://github.com/microsoft/DirectXTex
>     - https://walbourn.github.io/directx-tool-kit-for-audio-updates-and-a-direct3d-9-footnote/
> - https://walbourn.github.io/dxsetup-update/
> - https://walbourn.github.io/not-so-direct-setup/
> - https://walbourn.github.io/xinput-and-windows-8/

**Comment by Dwood15** [comment]
Yeah, I've done extensive googling on it. Good to have it documented, but a lot of that stuff doesn't map. 

Halo still uses DirectX9 under the hood, but those migration posts assume we're upgrading to modern DirectX. 

Re: Nuget, I want to avoid having a dependency on nuget for this project. And Microsoft has an arbitration clause on the License which is messed up. 

Rmoving the dep on directx extensions will potentially mean moving away from microsoft-provided libraries in favor of WINE-sourced ones, which will be more compatible with GPLv3.
